src/ios/PurchasesPlugin.swift

```diff
@@ -1,10 +1,397 @@
 //
 //  PurchasesPlugin.swift
-//  HelloCordova
+//  PurchasesPlugin
 //
-//  Created by Andrés Boedo on 9/23/20.
+//  Created by Joshua Liebowitz on 6/27/22
 //
-// This file is left intentionally blank,
 // the code for the plugin lives in CDVPurchasesPlugin
 
 import Foundation
+import PurchasesHybridCommon
+
+@objc public class CDVPurchasesPlugin : CDVPlugin {
+
+    private var updatedPurchaserInfoCallbackID: String?
+    private var shouldPurchasePromoProductCallbackID: String?
+    private var defermentBlocks: [RCDeferredPromotionalPurchaseBlock] = []
+
+    @objc(setupPurchases:) func setupPurchases(command: CDVInvokedUrlCommand) {
+        let apiKey = command.arguments[0] as? String ?? ""
+        let appUserID = command.arguments[1] as? String
+        let observerMode = command.arguments[2] as? NSNumber ?? NSNumber(value: false)
+        let userDefaultsSuiteName = command.arguments[3] as? String
+
+        Purchases.configure(withAPIKey: apiKey,
+                            appUserID: appUserID,
+                            observerMode: observerMode.boolValue,
+                            userDefaultsSuiteName: userDefaultsSuiteName,
+                            platformFlavor: self.platformFlavor,
+                            platformFlavorVersion: self.platformFlavorVersion)
+        Purchases.shared.delegate = self
+
+        self.updatedPurchaserInfoCallbackID = command.callbackId
+        let pluginResult = CDVPluginResult(status: .noResult)
+        pluginResult?.setKeepCallbackAs(true)
+        self.commandDelegate.send(pluginResult, callbackId: command.callbackId)
+    }
+
+    @objc(setAllowSharingStoreAccount:) func setAllowSharingStoreAccount(command: CDVInvokedUrlCommand) {
+        let allowSharingStoreAccount = command.arguments[0] as? NSNumber ?? NSNumber(value: false)
+
+        RCCommonFunctionality.setAllowSharingStoreAccount(allowSharingStoreAccount.boolValue)
+        self.sendOKFor(command: command)
+    }
+
+    @objc(addAttributionData:) func addAttributionData(command: CDVInvokedUrlCommand) {
+        let network = command.arguments[1] as? NSNumber
+        let networkUserId = command.arguments[2] as? NSString
+
+        guard let data = command.arguments[0] as? NSDictionary? as? [AnyHashable: Any],
+              let network = network?.intValue,
+              let networkUserId = networkUserId as? String else {
+            self.sendBadParametersFor(command: command,
+                                      parametersNamed: ["data", "network", "networkUserId"],
+                                      expectedTypes: [NSDictionary.self, NSNumber.self, NSString.self])
```

Comment:
I think it does since the underlying framework is objc, but I haven't tested. To lower the risk of messing things up, I stuck with objc things when it came to interfacing with the framework.

Reply:
Tested, it sends the objc types, but bridging does work, updated!

Reply:
awesome